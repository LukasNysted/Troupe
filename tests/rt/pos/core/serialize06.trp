(* -- POSITIVE -- added on 2021-06-12 by AA -- 

Positive test checking the interaction between serialization and built 
in functions. Built-in functions wrapped in a lambda should be serializable. 

*)


let val _ = save ("serialize06", fn () => random ) (* random is a built-in *)
    val f = restore ("serialize06")
    val g = f () 
    val r = g () 
in r >= 0 
end 



