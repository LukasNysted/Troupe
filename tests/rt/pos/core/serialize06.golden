2021-06-12T12:01:05.503Z [RTM] [32minfo[39m: Skipping network creation. Observe that all external IO operations will yield a runtime error.
>>> Main thread finished with value: true@{}%{}
